### PR TITLE
[FIX] website_sale_digital: allow download without reconciliation

### PR DESCRIPTION
**Pull request by yelizariev**
TLDR: after installation accounting app, invoices paid online get status "In Payment" until
reconciliation; this commit allows to download digital content without waiting
reconciliation which may take days or week

Steps:

* Install Accounting app
* Install website_sale_digital
* Create Digital product (Product Type: Service + use Digital Files smart button
to upload a file)
* Setup payment acquirer (e.g. Paypal)
* Buy digital product online and pay
* Open the Sale Order in Portal

Before:

* No "Downloads" button near to product name

After

* User can download files after purchasing

Notes

* In Odoo 13.0 it worked via "post_at" option, which is removed in Odoo 14. See
https://github.com/odoo/odoo/commit/caeb782841fc5a7ad71a196e2c9ee67644ef9074:

  > - Remove the post_at bank reconciliation feature. The "paid" state will be set on the invoices only if reconciled with a journal entry involving the journal's liquidity account.
  >     With invoicing, the payment will do that so the "in_payment" state should never be shown up.
  >     With accounting, only the statement lines have the power to move an invoice to the "paid" state.

---

opw-2353064

### CODE REVIEW

**Comment by yelizariev** [comment]
@nim-odoo

**Review by nim-odoo — COMMENTED** [review]
Does that mean that the user will have access to the product even if the invoice is not paid (e.g. in case of payment by a wire transfer?)

**Comment by yelizariev** [comment]
No. Invoice is not created with wire Transfer, so nothing will get "in payment" status

**Comment by yelizariev** [comment]
The portal page after wire transfer payment

image

**Comment by JKE-be** [comment]
in_payment = paid but not reconciled if I'm not wrong

**Comment by yelizariev** [comment]
It seems I accidentally requested review

image

**Comment by JKE-be** [comment]
as well you touch a website file, it is automatically added.
No accident here ;)

**Review by nim-odoo — APPROVED** [review]
> No. Invoice is not created with wire Transfer, so nothing will get "in payment" status

Indeed, you are right:

https://github.com/odoo/odoo/blob/ced500e9bcbb5c84033463689d135ad4c7dc4f39/addons/account/models/account_move.py#L1263

Hopefully there won't be an edge case with multi-currency + variable exchange rate.
This solution should be safe enough, though.

robodoo r+

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by Xavier-Do** [comment]
@robodoo retry
